### PR TITLE
Added a TTS adapter for the Qfrency speech engine from the CSIR.

### PR DESCRIPTION
**Pull request by rynkruger**
This commit adds the TTS adapter for the speech engine developed by CSIR South Africa, which we wish to contribute to the Daisy Pipeline project.

### CODE REVIEW

**Comment by bertfrees** [comment]
Nice. Thank you for the contribution!

I have quickly skimmed the code and I have couple of comments (see separate comments below).

What would also be great to have before merging this is:
- A README with some basic documentation about the engine itself (where to get it, etc.) and an example TTS configuration file
- List of well-known voices that we could possibly add to VoiceManager
- A minimal unit test

(I should also remember to update the [text-to-speech configuration page on our website](http://daisy.github.io/pipeline/modules/tts-common/tts-config.html).)

File: tts-utils/tts-adapter-qfrency/pom.xml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>org.daisy.pipeline.modules</groupId>
+    <artifactId>tts-modules-parent</artifactId>
+    <version>1.10.3-SNAPSHOT</version>
+    <relativePath>../../parent</relativePath>
+  </parent>
+
+  <version>3.0.1-SNAPSHOT</version>
```
Comment (bertfrees) [inline tts-utils/tts-adapter-qfrency/pom.xml]: Why did you choose 3.0.1? and not just 1.0.0?

File: tts-utils/tts-adapter-qfrency/pom.xml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+          <instructions>
+            <Service-Component>OSGI-INF/tts-service.xml</Service-Component>
+	    <Export-Package />
+          </instructions>
+        </configuration>
+      </plugin>
+    </plugins>
+  </build>
+
+  <profiles>
```
Comment (bertfrees) [inline tts-utils/tts-adapter-qfrency/pom.xml]: I assume this is a copy-paste left-over?

File: tts-utils/tts-adapter-qfrency/src/main/resources/transform-ssml.xsl [hunk 3]
```diff
@@ -0,0 +1,61 @@
+<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
```
Comment (bertfrees) [inline tts-utils/tts-adapter-qfrency/src/main/resources/transform-ssml.xsl]: It seems this files isn't used anywhere?

**Comment by rynkruger** [comment]
Re: unit tests, we were wondering how this should be implemented given that this is a commercial speech engine? How is it handeled for the other commercial engines? We thought of making a dummy client (that doesn't actually connect to a speech server) for running the unit tests. Would that be acceptable?

**Comment by bertfrees** [comment]
I believe for the other commercial engines it is handled by making the test optional. See for example tts-adapter-acapela. Without any special arguments passed to Maven, the tests are skipped.

Mocking the speech engine would certainly be acceptable. But if you can in addition provide a test that actually connects to the real engine (a integration test), that would be even better.

**Comment by bertfrees** [comment]
Thanks for the additions!

**Comment by rynkruger** [comment]
Hi, is there anything more required from me re: contribution of this driver?

**Comment by bertfrees** [comment]
No, I think it's complete. I will merge, and then this should normally be in the next release. Thanks again for your contribution.

**Comment by rynkruger** [comment]
Thank you very much for your help.

**Pull request merged**